subsys/net/ip/net_if.c

```diff
@@ -2820,19 +2820,25 @@ bool net_if_need_calc_rx_checksum(struct net_if *iface)
 
 struct net_if *net_if_get_by_index(u8_t index)
 {
-	if (&__net_if_start[index] >= __net_if_end) {
+	if (index == 0) {
+		return NULL;
+	}
+
+	if (&__net_if_start[index - 1] >= __net_if_end) {
 		NET_DBG("Index %d is too large", index);
 		return NULL;
 	}
 
-	return &__net_if_start[index];
+	return &__net_if_start[index - 1];
 }
 
 u8_t net_if_get_by_iface(struct net_if *iface)
```

Comment:
I guess type here needs to be changes to signed. Please just use `int`, that should lead to the mo[st/re] efficient code on RISCy 32-bit platforms.

Reply:
Yes, that probably makes sense to do now. I will send a fixed version.